test/declaration_test.ts

```diff
@@ -1174,6 +1177,36 @@ abstract class X {
   // Constructors on anonymous interfaces are not yet supported.
   /*external factory X(String|bool a, num b);*/
 }`);
+    });
+
+    it('should handle cases where a type and a variable have the same name but are unrelated',
+       () => {
+         // Case where a variable has the same name as a type, but they should not be combined
+         // because they are unrelated. This is valid in TypeScript because the variable has an
+         // actual value in JavaScript, whereas the type is only part of TypeScript's type system.
+         expectTranslate(`
+interface X {
+  a: string;
+  b: number;
+  c(): boolean;
+}
+
+declare var X: { a: number[], b: number[], c: number[] };
+`).to.equal(`@anonymous
+@JS()
+abstract class XInterface {
```

Comment:
to make this test better also reference an object of type X and make sure it is given type XInterface not X.